dotnet/src/Connectors/Connectors.Memory.Postgres/PostgresVectorStoreCollectionSqlBuilder.cs

```diff
@@ -0,0 +1,447 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Microsoft.Extensions.VectorData;
+using Npgsql;
+using NpgsqlTypes;
+using Pgvector;
+
+namespace Microsoft.SemanticKernel.Connectors.Postgres;
+
+/// <summary>
+/// Provides methods to build SQL commands for managing vector store collections in PostgreSQL.
+/// </summary>
+internal class PostgresVectorStoreCollectionSqlBuilder : IPostgresVectorStoreCollectionSqlBuilder
+{
+    /// <inheritdoc />
+    public PostgresSqlCommandInfo BuildDoesTableExistCommand(string schema, string tableName)
+    {
+        return new PostgresSqlCommandInfo(
+            commandText: $@"
+                SELECT table_name
+                FROM information_schema.tables
+                WHERE table_schema = $1
+                    AND table_type = 'BASE TABLE'
+                    AND table_name = '{tableName}'",
+            parameters: [new NpgsqlParameter() { Value = schema }]
+        );
+    }
+
+    /// <inheritdoc />
+    public PostgresSqlCommandInfo BuildGetTablesCommand(string schema)
+    {
+        return new PostgresSqlCommandInfo(
+            commandText: @"
+                SELECT table_name
+                FROM information_schema.tables
+                WHERE table_schema = $1
+                    AND table_type = 'BASE TABLE'",
+            parameters: [new NpgsqlParameter() { Value = schema }]
+        );
+    }
+
+    /// <inheritdoc />
+    public PostgresSqlCommandInfo BuildCreateTableCommand(string schema, string tableName, IReadOnlyList<VectorStoreRecordProperty> properties, bool ifNotExists = true)
+    {
+        if (string.IsNullOrWhiteSpace(tableName))
+        {
+            throw new ArgumentException("Table name cannot be null or whitespace", nameof(tableName));
+        }
+
+        VectorStoreRecordKeyProperty? keyProperty = default;
+        List<VectorStoreRecordDataProperty> dataProperties = new();
+        List<VectorStoreRecordVectorProperty> vectorProperties = new();
+
+        foreach (var property in properties)
+        {
+            if (property is VectorStoreRecordKeyProperty keyProp)
+            {
+                if (keyProperty != null)
```

Comment:
FYI, since you constructed the property reader with the flat that multiple keys are not supported, it would have thrown already at collection construction time if multiple key properties existed.